### DISCUSSION TITLE
如何将图标改用阿里云的Iconfont

### DISCUSSION THREAD

**Discussion by goldsheep3** [ppoffice/hexo-theme-icarus#929] [Q&A] [answered]
主题应用之后，在Font Awesome图标库中没有找到所需要的图标，便想更换为[Iconfont](https://www.iconfont.cn/)的图标库。然而，尝试的结果均失败。

尝试了以下方式：
- 在meta中加入：
  - `<link rel="stylesheet" href="//at.alicdn.com/t/font_8d5l8fzk5b87iudi.css">`
  - `<link rel="stylesheet" href="https://at.alicdn.com/t/font_8d5l8fzk5b87iudi.css">`
  - `<link rel="stylesheet" href="\css\iconfont.css">`（此时换用了本地文件方式安装主题，而非npm）
- 设置图标CDN为https://at.alicdn.com/t/font_8d5l8fzk5b87iudi.css
- 在icon部分切换调用iconfont icon-huoban88与`<i class="iconfont icon-huoban88"></i>`

仍然尚未解决问题，在尝试过后网页不显示任何图标。是否有办法解决这一问题？

**Comment by ppoffice** [comment, marked as answer]
@goldsheep3 如果图标元素的父元素有`is-transparent`的classname，则移除`is-transparent`。

**Reply by goldsheep3** [reply]
并未找到`is-transparent`的类名......我会将目前的文件存储到仓库goldsheep3/goldsheep3.github.io的分支dev中，是否可劳烦您看一下？

**Reply by ppoffice** [reply]
把下面这行去掉试试：
https://github.com/ppoffice/hexo-theme-icarus/blob/1e563d6bd8fd1f1e377a1cc247cacc28c962200d/include/style/button.styl#L6

**Reply by ppoffice** [reply]
你的配置文件也不对呀，https://at.alicdn.com/t/font_8d5l8fzk5b87iudi.css 中哪里有`icon-Discord`？而且`icon: `后面填的是classname，`<i class... />`是没有用的。
https://github.com/goldsheep3/goldsheep3.github.io/blob/dev/_config.icarus.yml#L240-L255

**Reply by goldsheep3** [reply]
啊明白了明白了明白了……是我尝试的时候把iconfont自己生成的css和示例css搞混了，竟然没发现
很感谢！

**Reply by goldsheep3** [reply]
另外刚刚发现的问题，切换成iconfont后「捐赠」和「许可协议」部分的图标仍然是fontawesome的图标，而且没有找到修改的方法，这是否可以修改？

**Reply by ppoffice** [reply]
> 另外刚刚发现的问题，切换成iconfont后「捐赠」和「许可协议」部分的图标仍然是fontawesome的图标，而且没有找到修改的方法，这是否可以修改？

许可协议图标也是在_config.yml里面有设置的。捐赠按钮就只有改代码了

**Reply by goldsheep3** [reply]
许可协议已经在`_config.yml`找到了:joy:不过捐赠按钮的图标我翻遍了整个代码都没有找到其他的图标了（是用Atom的搜索功能搜索关键词`fa-`，整个theme文件夹已经全部搜索过了）

**Reply by ppoffice** [reply]
在另一个npm package里
https://github.com/ppoffice/hexo-component-inferno
